demos/defender/defender_demo_json/CMakeLists.txt

```diff
@@ -42,9 +42,9 @@ set_macro_definitions(TARGETS ${DEMO_NAME}
                         "ROOT_CA_CERT_PATH"
                         "CLIENT_CERT_PATH"
                         "CLIENT_PRIVATE_KEY_PATH"
-                        "CLIENT_IDENTIFIER"
                         "THING_NAME"
```

Comment:
As part of best practices for connection with IoT Core, the Client ID should be the same as the Thing Name. 
So, should we make the CLIENT_IDENTIFIER as the required config but THING_NAME as the optional config?

Reply:
Required in the context of this function means that a definition is required for it to compile without error. In that case, I think both of these values should be required. In fact, nothing in the defender demo should be optional.